packages/go_router/test/custom_transition_page_test.dart

```diff
@@ -84,6 +84,38 @@ void main() {
       homeScreenPositionInTheMiddleOfRemoval,
     );
   });
+
+  testWidgets('Dismiss a screen by tapping a modal barrier',
+      (WidgetTester tester) async {
+    const ValueKey<String> homeKey = ValueKey<String>('home');
+    const ValueKey<String> dismissibleModalKey =
+        ValueKey<String>('dismissibleModal');
+
+    final GoRouter router = GoRouter(
+      routes: <GoRoute>[
+        GoRoute(
+          path: '/',
+          builder: (_, __) => const HomeScreen(key: homeKey),
+        ),
+        GoRoute(
+          path: '/dismissible-modal',
+          pageBuilder: (_, __) => CustomTransitionPage<void>(
+            barrierDismissible: true,
```

Comment:
a extreme nitpicking, lets make sure we assign state.key to the Page.key, in case, people copy paste this code and run into issue